src/components/ha-bar-slider.ts

```diff
@@ -0,0 +1,330 @@
+        break;
+      case "PageUp":
+        this.value = this.steppedValue(
+          this.boundedValue(this.value ?? 0 + this._tenPercentStep)
+        );
+        break;
+      case "PageDown":
+        this.value = this.steppedValue(
+          this.boundedValue(this.value ?? 0 - this._tenPercentStep)
+        );
+        break;
+      case "Home":
+        this.value = this.min;
+        break;
+      case "End":
+        this.value = this.max;
+        break;
+    }
+    fireEvent(this, "slider-moved", { value: this.value });
+  }
+
+  _handleKeyUp(e: KeyboardEvent) {
+    if (!A11Y_KEY_CODES.has(e.code)) return;
+    e.preventDefault();
+    fireEvent(this, "value-changed", { value: this.value });
+  }
+
+  destroyListeners() {
+    if (this._mc) {
+      this._mc.destroy();
+      this._mc = undefined;
+    }
+  }
+
+  protected render(): TemplateResult {
+    return html`
+      <div
+        class=${classMap({
+          container: true,
+          controlled: this.controlled,
+        })}
+      >
+        <div
+          id="slider"
+          class="slider"
+          style=${styleMap({
+            "--value": `${this.valueToPercentage(this.value ?? 0)}`,
+          })}
+          tabindex="0"
+          @keydown=${this._handleKeyDown}
+          @keyup=${this._handleKeyUp}
+          role="slider"
+          aria-valuemin=${this.min}
+          aria-valuemax=${this.max}
+          aria-valuenow=${this.value ?? 0}
+          aria-labelledby=${ifDefined(this.label)}
+        >
+          <div class="slider-track-background"></div>
+          ${this.trackMode === "active"
+            ? html`<div class="slider-track-active"></div>`
+            : null}
+          ${this.trackMode === "indicator"
+            ? html`<div class="slider-track-indicator"></div>`
+            : null}
+        </div>
+      </div>
+    `;
+  }
+
+  static get styles(): CSSResultGroup {
+    return css`
+      :host {
+        --main-color: rgba(var(--rgb-primary-color), 1);
+        --bg-gradient: none;
+        --bg-color: rgba(var(--rgb-secondary-text-color), 0.2);
+      }
+      .container {
+        display: flex;
+        flex-direction: row;
+        height: 40px;
+      }
+      .slider {
+        position: relative;
+        height: 100%;
+        width: 100%;
+        border-radius: 16px;
+        transform: translateZ(0);
+        overflow: hidden;
+        cursor: pointer;
+      }
+      .slider * {
+        pointer-events: none;
+      }
+      .slider .slider-track-background {
+        position: absolute;
+        top: 0;
+        left: 0;
+        height: 100%;
+        width: 100%;
+        background-color: var(--bg-color);
+        background-image: var(--gradient);
+      }
+      .slider .slider-track-active {
+        position: absolute;
+        top: 0;
+        left: 0;
+        height: 100%;
+        width: 100%;
+        transform: translate3d(calc(-100% + var(--value, 0) * 100%), 0, 0);
+        border-radius: 0 8px 8px 0;
```

Comment:
Is it by design that the slider has a different border radius then the track? Looks a bit weird...

Reply:
Yep 😅
![image](https://user-images.githubusercontent.com/5878303/196125181-f94fe961-eb60-4de0-b567-08f8c6f37a7e.png)